### PR TITLE
test: unmount the mountpoint just before exiting

### PR DESCRIPTION
**Pull request by lxbsz**
Without this the qa test may fail by evicting the unresponsive client after 300 seconds.

Fixes: https://tracker.ceph.com/issues/61394

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by kotreshhr — APPROVED** [review]

**Comment by dparmar18** [comment]
By this patch you mean to say that the second process (non zero id) and thus didn't get unmounted and so by replicating the `if` stmt, things should be good. If so better would be be keep `ceph_unmount(cmount);` after `if-else` i.e just before return stmt.

**Comment by neesingh-rh** [comment]
> By this patch you mean to say that the second process (non zero id) and thus didn't get unmounted and so by replicating the `if` stmt, things should be good. If so better would be be keep `ceph_unmount(cmount);` after `if-else` i.e just before return stmt.

Agree, if that's the case

> By this patch you mean to say that the second process (non zero id) and thus didn't get unmounted and so by replicating the `if` stmt, things should be good. If so better would be be keep `ceph_unmount(cmount);` after `if-else` i.e just before return stmt.

Agree!! If that's the case , it would be good to have a common `ceph_unmount(cmount)` just before the return stmt.

**Comment by neesingh-rh** [comment]
nit: I guess you mean `exiting` instead of `existing` in the PR name.

**Comment by lxbsz** [comment]
@dparmar18 @neesingh-rh

Makes sense to me and fixed it, thanks!

File: src/test/fs/test_trim_caps.cc [hunk 1]
```diff
@@ -78,8 +77,10 @@ int main(int argc, char *argv[])
 		ret = ceph_statx(cmount, ".", &stx, 0, 0);
 		assert(ret >= 0);
 
-		printf("waiting for crash\n");
-        sleep(60);
+		printf("waiting for 60 seconds to see whether will it crash\n");
+		sleep(60);
+		printf("parent exits\n");
```
Comment (dparmar18) [inline src/test/fs/test_trim_caps.cc:82]: Will process with non-zero ID be parent? Shouldn't this be child?

Comment (lxbsz) [reply src/test/fs/test_trim_caps.cc:82]: The `fork()` will return `0` in the child process and the child pid in the parent process, more detail please see the `man fork`:

```
RETURN VALUE
       On success, the PID of the child process is returned in the parent, and 0 is returned in the child.  On failure, -1 is returned in the parent, no child process is created, and errno is set to indicate the error.

```

**Review by dparmar18 — APPROVED** [review]
LGTM

**Comment by dparmar18** [comment]
jenkins test api

**Review by neesingh-rh — APPROVED** [review]

**Review by joscollin — APPROVED** [review]

**Review by rishabh-d-dave — COMMENTED** [review]
https://tracker.ceph.com/projects/cephfs/wiki/Main#22-Sep-2023

**Review by rishabh-d-dave — APPROVED** [review]

**Comment by rishabh-d-dave** [comment]
jenkins test windows

**Comment by rishabh-d-dave** [comment]
I'll merge after this test passes (since it's not possible to merge the PR without it).

**Comment by rishabh-d-dave** [comment]
Error looks unrelated - https://jenkins.ceph.com/job/ceph-windows-pull-requests/29084/consoleFull#-213060095940526d21-3511-427d-909c-dd086c0d1034

```
++ echo 'Timeout waiting for the VM to start'
Timeout waiting for the VM to start
++ return 1
```

Trying one more time.

**Pull request merged**